io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -578,7 +597,31 @@ public Instance next() {
 			}
 
 			Instance instance = iterator.next();
-			return new StreamGmlInstance(instance, totalFeaturesProcessed++);
+
+			Iterable<QName> mapPropertiess = instance.getPropertyNames();
+			for (QName qName : mapPropertiess) {
+				if (qName.getLocalPart().equals("id") && qName.getPrefix().equals("gml")) {
+					Object[] gmlID = instance.getProperty(qName);
+					String gmlIDToCheck = (String) gmlID[0];
```

Comment:
There should be a `null` check, also rather than risking a `ClassCastException` use `toString()`.